### PR TITLE
Adds tiny circuits

### PR DESCRIPTION
**Pull request by Hate9**
This adds tiny circuits to the game (and ensures that circuit max values are rounded, since that apparently wasn't a thing before).
There are five different versions of these circuits (all entirely cosmetic differences).

:cl: Hate9
add: Added tiny-sized circuits (called Devices)
imageadd: added new icons for the Devices
/:cl:

[why]: It seemed like there should be tiny circuits, so you can make really simple ones take up even less space, and fit in new places.

### CODE REVIEW

**Comment by JJRcop** [comment]
#38621

**Comment by Hate9** [comment]
In that case, I guess I should make an exception to grabbers for these, since they can't actually hold throwers.

**Comment by JJRcop** [comment]
I think these tiny circuits are great, but unfortunately I don't support your reason adding them

**Comment by Hate9** [comment]
Why shouldn't they? The problem with picking circuits up (from my understanding) was that you could have weapons-in-weapons, wasn't it?

**Comment by Hate9** [comment]
Also, can someone more well-versed in this explain to me what went wrong with the travis-ci build? I don't quite get it.

**Comment by JJRcop** [comment]
One of our scripts tries to access html/changelogs/example.yml, and you accidentally renamed it

**Comment by Hate9** [comment]
Oh, good to know.

**Comment by Hate9** [comment]
Fixed, for the record. No idea how that happened.

**Comment by Tlaltecuhtli** [comment]
can you make tiny mobile circuits?

**Comment by Hate9** [comment]
I almost certainly will at some point, but it sounds somewhat less likely to be accepted, so I'll probably make it in a separate PR.

Plus, that requires more spriting.

**Comment by Hate9** [comment]
Actually, what would the point of tiny mobile circuits be?

**Comment by Garen7** [comment]
Mobile circuits weren't meant to be fit into storage and also the complexity cap is really low on tiny circuits so you'd struggle if you were trying to make anything with them.

**Comment by Hate9** [comment]
I feel like that does the opposite of answer my question.

**Comment by Garen7** [comment]
Oh, I wasn’t the person asking you to make them, I was making a point against them

**Comment by Iamgoofball** [comment]
Nested circuits are fine.

**Comment by Hate9** [comment]
Nested circuits break the intended limits of circuits.
I'm working on another PR to fix that, too, though.

**Comment by Borisvanmemes** [comment]
The "hook" case looks a lot like a pen, being able to put it into a PDA would be cool

**Comment by Hate9** [comment]
I wonder if that would be easy to do...

File: html/changelogs/tinycircuits.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: "Hate9"
```
Comment (ShizCalev) [inline html/changelogs/tinycircuits.yml]: Remove this file please, it's automatically generated based off your PR's comment.

**Review by ShizCalev — DISMISSED** [review]

**Comment by Hate9** [comment]
Removed the extra changelog file.

**Comment by Hate9** [comment]
@ShizCalev Done.

**Comment by Hate9** [comment]
Bump.

**Pull request merged**